lib/rubocop/ast/traversal.rb

```diff
@@ -14,170 +14,38 @@ def walk(node)
         nil
       end
 
-      NO_CHILD_NODES    = %i[true false nil int float complex
-                             rational str sym regopt self lvar
-                             ivar cvar gvar nth_ref back_ref cbase
-                             arg restarg blockarg shadowarg
-                             kwrestarg zsuper lambda redo retry].freeze
-      ONE_CHILD_NODE    = %i[splat kwsplat block_pass not break next
-                             preexe postexe match_current_line defined?
-                             arg_expr].freeze
-      MANY_CHILD_NODES  = %i[dstr dsym xstr regexp array hash pair
-                             mlhs masgn or_asgn and_asgn
-                             undef alias args super yield or and
-                             while_post until_post iflipflop eflipflop
-                             match_with_lvasgn begin kwbegin return].freeze
-      SECOND_CHILD_ONLY = %i[lvasgn ivasgn cvasgn gvasgn optarg kwarg
-                             kwoptarg].freeze
+      NO_CHILD_NODES   = %i[arg back_ref blockarg cbase complex const cvar
+                            false float gvar int ivar kwrestarg lambda lvar nil
+                            nth_ref rational redo regopt restarg retry self
+                            shadowarg str sym true zsuper].freeze
+      SOME_CHILD_NODES = %i[alias and and_asgn arg_expr args array begin block
+                            block_pass break case casgn class csend cvasgn def
+                            defined? defs dstr dsym eflipflop ensure erange for
+                            gvasgn hash if iflipflop irange ivasgn kwarg
+                            kwbegin kwoptarg kwsplat lvasgn masgn
+                            match_current_line match_with_lvasgn mlhs module
+                            next not op_asgn optarg or or_asgn pair postexe
+                            preexe regexp resbody rescue return sclass send
+                            splat super undef until until_post when while
+                            while_post xstr yield].freeze
 
       NO_CHILD_NODES.each do |type|
         module_eval("def on_#{type}(node); end", __FILE__, __LINE__)
       end
 
-      ONE_CHILD_NODE.each do |type|
+      SOME_CHILD_NODES.each do |type|
         module_eval(<<-RUBY, __FILE__, __LINE__ + 1)
           def on_#{type}(node)
-            if (child = node.children[0])
-              send(:"on_\#{child.type}", child)
-            end
-          end
-        RUBY
-      end
+            node.children.each do |child|
```

Comment:
You could try `#each_child_node` here instead of `#children` + `#each`. It should ignore all non-node children.